### PR TITLE
Regenerating clients (breaking changes)

### PR DESCRIPTION
**Pull request by garrettjonesgoogle**
This is a large regeneration to clean up the surface of generated clients. 

Corresponds to https://github.com/googleapis/toolkit/pull/1659 and depends on https://github.com/googleapis/gax-java/pull/396 being released.

### CODE REVIEW

File: google-cloud-core-grpc/src/main/java/com/google/cloud/grpc/BaseGrpcServiceException.java [hunk 1]
```diff
@@ -72,8 +72,8 @@ public BaseGrpcServiceException(ApiException apiException) {
         .setMessage(apiException.getMessage())
         .setCause(apiException)
         .setRetryable(apiException.isRetryable())
-        .setCode(((GrpcStatusCode) apiException.getStatusCode()).getCode().value())
-        .setReason(((GrpcStatusCode) apiException.getStatusCode()).getCode().name())
+        .setCode(apiException.getStatusCode().getCode().getHttpStatusCode())
```
Comment (pongad) [inline google-cloud-core-grpc/src/main/java/com/google/cloud/grpc/BaseGrpcServiceException.java]: I might be missing something. Why is this HTTP code? This is for gRPC right?

File: google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/Subscriber.java [hunk 2]
```diff
@@ -168,11 +171,18 @@ public void close() throws IOException {
...
+    }
+    if (channelProvider.needsHeaders()) {
+      channelProvider = channelProvider.withHeaders(builder.headerProvider.getHeaders());
+    }
+    this.channelProvider = channelProvider;
     credentialsProvider = builder.credentialsProvider;
 
     numChannels = builder.parallelPullCount;
-    channels = new ArrayList<ManagedChannel>(numChannels);
+    channels = new ArrayList(numChannels);
```
Comment (pongad) [inline google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/Subscriber.java]: `new ArrayList<>(numChannels)`

File: google-cloud-pubsub/src/test/java/com/google/cloud/pubsub/v1/PublisherImplTest.java [hunk 3]
```diff
@@ -16,40 +16,39 @@
...
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
 import org.threeten.bp.Duration;
 
+import java.io.IOException;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ScheduledExecutorService;
+
+import static org.junit.Assert.*;
```
Comment (pongad) [inline google-cloud-pubsub/src/test/java/com/google/cloud/pubsub/v1/PublisherImplTest.java]: Which formatting are we using for this repo? This doesn't conform to Google style guide; the style guide doesn't allow spaces between imports.

File: google-cloud-core-grpc/src/main/java/com/google/cloud/grpc/BaseGrpcServiceException.java (hunk 1 above)
Comment (garrettjonesgoogle) [reply google-cloud-core-grpc/src/main/java/com/google/cloud/grpc/BaseGrpcServiceException.java]: Yes, but the base class `BaseServiceException` expects an integer code. It was previously using the proto field number for grpc exceptions, but I don't think that was useful to anyone.

File: google-cloud-pubsub/src/test/java/com/google/cloud/pubsub/v1/PublisherImplTest.java (hunk 3 above)
Comment (garrettjonesgoogle) [reply google-cloud-pubsub/src/test/java/com/google/cloud/pubsub/v1/PublisherImplTest.java]: Ack that was an IDE problem, I'll fix it.

**Comment by pongad** [comment]
This LGTM, but I think at least one more person should approve.

**Pull request merged**